.github/workflows/c.yaml

```diff
@@ -12,38 +12,34 @@ on:
       - "c/**"
       - .github/workflows/c.yaml
 jobs:
-
   ubuntu-lts:
-
     strategy:
       matrix:
-        build_type: [ "Debug", "Release" ]
-        compiler: [
-          { c: "gcc-6", cxx: "g++-6", package: "gcc-6 g++-6" },
-          { c: "clang-6.0", cxx: "clang++-6.0", package: "clang-6.0" }
-        ]
-        build_shared_libraries: [ true, false ]
-        exclude: [
-          { build_type: "Debug", compiler: { c: "gcc-6", cxx: "g++-6" }, build_shared_libraries: false }
-        ]
+        build_type: ["Debug", "Release"]
+        compiler:
+          [
+            { c: "gcc-14", cxx: "g++-14", package: "gcc-14 g++-14" },
+            { c: "clang-18", cxx: "clang++-18", package: "clang-18" },
+          ]
+        build_shared_libraries: [true, false]
 
-    name: "Ubuntu 18.04 (Build: ${{ matrix.build_type }}, Compilers: ${{ matrix.compiler.c }}/${{ matrix.compiler.cxx }}, Shared Library: ${{ matrix.build_shared_libraries }})"
+    name: "Ubuntu 24.04 (Build: ${{ matrix.build_type }}, Compilers: ${{ matrix.compiler.c }}/${{ matrix.compiler.cxx }}, Shared Library: ${{ matrix.build_shared_libraries }})"
 
-    runs-on: ubuntu-latest
-    container: ubuntu:18.04
+    runs-on: ubuntu-24.04
```

Comment:
I would say we shouldn't run on the Ubuntu 24 instance, we should always use a container within the runner, the runner's version can always be the latest. The reason for this is to avoid finding ourselves in a position where Github no longer supports a version of a runner and our builds breaks, which happens regularly enough. We you could argue that the container would suffer similar issues as we always calls the `apt-get` to pull in packages from a remote source but the remote source support things for many years, we are still running 18.04 without a problem (apart from the github action which recently was fixed up).

Reply:
Roger that, corrected